src/components/Chat/hooks/useStreami18n.ts

```diff
@@ -1,18 +1,25 @@
 import { useEffect } from 'react';
 
+import type { TranslationContextValue } from '../../../contexts/translationContext/TranslationContext';
 import { Streami18n } from '../../../utils/Streami18n';
 
-export const useStreami18n = ({ i18nInstance, setTranslators }) => {
+export const useStreami18n = ({
+  i18nInstance,
+  setTranslators,
+}: {
+  setTranslators: React.Dispatch<React.SetStateAction<TranslationContextValue>>;
+  i18nInstance?: typeof Streami18n;
+}) => {
   useEffect(() => {
-    let streami18n;
+    let streami18n: Streami18n;
 
     if (i18nInstance instanceof Streami18n) {
       streami18n = i18nInstance;
     } else {
       streami18n = new Streami18n({ language: 'en' });
     }
 
-    streami18n.registerSetLanguageCallback((t) =>
+    streami18n.registerSetLanguageCallback((t: (key: string) => string) =>
```

Comment:
this `t` function type something we want to export from Streami18n cause we'll be using it in a good amount of places?